chiselFrontend/src/main/scala/chisel3/Data.scala

```diff
@@ -712,34 +729,35 @@ object WireDefault {
   }
 }
 
-/** RHS (source) for Invalidate API.
-  * Causes connection logic to emit a DefInvalid when connected to an output port (or wire).
-  */
-private[chisel3] object DontCare extends Element {
-  // This object should be initialized before we execute any user code that refers to it,
-  //  otherwise this "Chisel" object will end up on the UserModule's id list.
-  // We make it private to chisel3 so it has to be accessed through the package object.
+package internal {
+  /** RHS (source) for Invalidate API.
+    * Causes connection logic to emit a DefInvalid when connected to an output port (or wire).
+    */
+  private[chisel3] object InternalDontCare extends Element {
```

Comment:
Can this not be internal (object aliasing is kind of what we're trying to avoid right)? Is there a reason it needs to be accessed through the package object?

Reply:
We need to ensure the object is created before it's reference by user code.
My earlier attempt at doing so (`initializeSingletons()`) causes random deadlock during testing with multiple threads.